### PR TITLE
creds from env, 1sec wait_timeout

### PR DESCRIPTION
**Pull request by slorello89**

### CODE REVIEW

File: src/redis_cache.cc [hunk 1]
```diff
@@ -118,6 +127,11 @@ RedisCache::Create(
   sw::redis::ConnectionOptions options;
   sw::redis::ConnectionPoolOptions poolOptions;
 
+  // try pulling user/password from environment fist
+  // override if present in the config
+  setOptionFromEnv(USERNAME_ENV_VAR_NAME, options.user);
```
Comment (Spartee) [inline src/redis_cache.cc:132]: Might as well just add in TLS cert/pem file location as option and support TLS communication? SSL?

**Review by Spartee — COMMENTED** [review]
Just one comment, otherwise seems good to me.

File: src/redis_cache.cc (hunk 1 above)
Comment (slorello89) [reply src/redis_cache.cc:132]: Opened #7 to address just this, not sure they really need to be accessible in the env vars, doing that for user/password as those credentials can be sensitive, for TLS you're really just passing in file names.

**Review by rmccorm4 — COMMENTED** [review]
LGTM, but I added a [test](https://github.com/triton-inference-server/server/pull/5966). Will wait to check its passing before approving.

File: README.md [hunk 2]
```diff
@@ -93,17 +93,21 @@ tritonserver --cache-config redis,host=redis-host --cache-config redis,port=6379
...
+| db | No | The db number to user. NOTE - use of the db number is considered an anti-pattern in Redis, so it is advised that you do not use this option | 0       |
+| connect_timeout | No | The maximum time, in milliseconds to wait for a connection to be established to Redis. 0 means wait forever                                 | 0       |
+| socket_timeout | No | The maximum time, in milliseconds the client will wait for a response from Redis. 0 means wait forever                                      | 0       |
+| pool_size | No | The number pooled connections to Redis the client will maintain.                                                                            | 1       |
+| wait_timeout | No | The maximum time, in milliseconds to wait for a connection from the pool.                                                                   | 1000    |
 
 
+### Optional Environment Variables for Credentials
+
+Optionally you may configure your `user`/`password` via environment variables. The corresponding `user` environment variable is `TRITONCACHE_REDIS_USERNAME` whereas the corresponding password environment variable is `TRITONCACHE_REDIS_PASSWORD`.
```
Comment (rmccorm4) [inline README.md:109]:
```suggestion
Optionally you may configure your `user`/`password` via environment variables. The corresponding `user` environment variable is `TRITONCACHE_REDIS_USERNAME` whereas the corresponding `password` environment variable is `TRITONCACHE_REDIS_PASSWORD`.
```

**Review by rmccorm4 — APPROVED** [review]
Test passes, LGTM

**Pull request merged**